pydis_site/apps/api/views.py

```diff
@@ -226,3 +231,100 @@ def get(
                 "error": str(e),
                 "requested_resource": f"{owner}/{repo}/{sha}/{action_name}/{artifact_name}"
             }, status=e.status)
+
+
+class GitHubWebhookFilterView(APIView):
+    """
+    Filters uninteresting events from webhooks sent by GitHub to Discord.
+
+    ## Routes
+    ### POST /github/webhook-filter/:webhook_id/:webhook_token
+    Takes the GitHub webhook payload as the request body, documented on here:
+    https://docs.github.com/en/webhooks/webhook-events-and-payloads. The endpoint
+    will then determine whether the sent webhook event is of interest,
+    and if so, will forward it to Discord. The response from Discord is
+    then returned back to the client of this website, including the original
+    status code and headers (excluding `Content-Type`).
+
+    ## Authentication
+    Does not require any authentication nor permissions on its own, however,
+    Discord will validate that the webhook originates from GitHub and respond
+    with a 403 forbidden error if not.
+    """
+
+    authentication_classes = ()
+    permission_classes = ()
+
+    def post(self, request: Request, *, webhook_id: str, webhook_token: str) -> Response:
+        """Filter a webhook POST from GitHub before sending it to Discord."""
+        sender = request.data.get('sender', {}).get('login', '')
+        event = request.headers.get('X-GitHub-Event')
+
+        is_coveralls = 'coveralls' in sender
+        is_github_bot = '[bot]' in sender
+        is_sentry = 'sentry-io' in sender
+        is_dependabot_branch_delete = (
+            'dependabot' in request.data.get('ref', '')
+            and event == 'delete'
+        )
+        is_bot_pr_approve = (
+            '[bot]' in request.data.get('pull_request', {}).get('user', {}).get('login', '')
+            and event == 'pull_request_review'
+        )
+        is_empty_review = (
+            request.data.get('review', {}).get('state') == 'commented'
```

Comment:
It would be better to reduce the verbosity of all of these dictionary accesses we're doing by assing them to variables.

e.g. `review = request.data.get('review', {})`, etc, etc.

Reply:
I've changed this for some things that were used multiple times, but e.g. `ref` or `pull_request` are only used once, so I'm not sure it's worth it?

Reply:
Sure, we want to avoid repetition, so if it's not repeated, it's fine.